lib/cadet/assessments/assessments.ex

```diff
@@ -481,6 +506,112 @@ defmodule Cadet.Assessments do
     Question.changeset(%Question{}, params_with_assessment_id)
   end
 
+  def insert_voting(
+        contest_number,
+        usernames,
+        question_id
+      ) do
+    contest_assessment = Repo.get_by(Assessment, number: contest_number)
+    users = Enum.map(usernames, fn x -> Repo.get_by(User, username: x.username) end)
+    invalid_usernames = Enum.any?(users, fn user -> user == nil end)
+
+    if contest_assessment != nil and !invalid_usernames do
+      contest_submission_ids =
+        Submission
+        |> where(assessment_id: ^contest_assessment.id)
+        |> where(status: "submitted")
+        |> Repo.all()
+        |> Enum.map(fn x -> x.id end)
```

Comment:
Please use select() rather than selecting everything from the database and then mapping it to only the `id`...